pandas/io/excel.py

```diff
@@ -824,8 +824,56 @@ class ExcelWriter(object):
 
     Notes
     -----
+    None of methods and properties are considered public.
+
     For compatibility with CSV writers, ExcelWriter serializes lists
     and dicts to strings before writing.
+
+    Examples
+    --------
+    Using ExcelWriter, some settings can be added.
+
+    Default usage.
+
+    >>> with ExcelWriter('path_to_file.xlsx') as writer:
+    ...     df.to_excel(writer)
+
+    If you want to set engine that can manipulate Excel,
+    pass keyword argument named engine. Actually
+    engine is automatically chosen by file extension.
+
+    >>> with ExcelWriter('path_to_file.xlsx', engine='openpyxl') as writer:
+    ...     df.to_excel(writer)
+
+    In order to write separate DataFrames to separate sheets
+    in a single Excel file, one can pass an ExcelWriter.
+
+    >>> with ExcelWriter('path_to_file.xlsx') as writer:
+    ...     df1.to_excel(writer, sheet_name='Sheet1')
+    ...     df2.to_excel(writer, sheet_name='Sheet2')
+
+    You can set date format or datetime format
+
+    >>> with ExcelWriter('path_to_file.xlsx',
```

Comment:
I am actually surprised that we allow date / date time formatting with ExcelWriter but don't have the same usage in `to_excel` - if you are up to it aligning that functionality would be ideal (separate change)